Sources/Brave/Frontend/Browser/PrivacyHub/PrivacyReportsManager.swift

```diff
@@ -19,8 +19,8 @@ public struct PrivacyReportsManager {
   /// Instead a periodic timer is run and all requests gathered during this timeframe are saved in one database transaction.
   public static var pendingBlockedRequests: [(host: String, domain: URL, date: Date)] = []
   
-  public static func processBlockedRequests() {
-    if PrivateBrowsingManager.shared.isPrivateBrowsing { return }
+  public static func processBlockedRequests(isPrivateBrowsing: Bool) {
+    if isPrivateBrowsing { return }
```

Comment:
Ahh they were all originally happening inside these functions, and all I did was change it to use the one that was passed in, instead of the singleton `Preference`.

I can change it to the call-site yeah. It's just that in so many places I had to change from all the `Preferences.Privacy.privateBrowsing.value` stuff being embedded in legit every function :(

I basically did this change everywhere, passing it in, as a sort of simplified way of handling it, without changing everything TOO drastically.

But if we're okay with me changing it to the call-site EVERYWHERE, then I'm fine with that too.

Reply:
Fixed